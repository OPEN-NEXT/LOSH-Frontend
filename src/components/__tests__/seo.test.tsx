import * as React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { useStaticQuery } from "gatsby";
import site from "../../../gatsby-config";
import SEO from "../seo";

const renderer = ShallowRenderer.createRenderer();

beforeEach(() => {
	useStaticQuery.mockImplementationOnce(() => ({
		site
	}));
});

describe("SEO", () => {
	it("renders the test correctly", () => {
		const tree = renderer.render(<SEO title="All posts" />);

		expect(tree).toMatchSnapshot();
	});
});
