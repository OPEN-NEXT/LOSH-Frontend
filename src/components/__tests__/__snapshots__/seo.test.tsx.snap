// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders the test correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": undefined,
    }
  }
  meta={
    Array [
      Object {
        "content": "TypeScript version of the default Gatsby starter",
        "name": "description",
      },
      Object {
        "content": "All posts",
        "property": "og:title",
      },
      Object {
        "content": "TypeScript version of the default Gatsby starter",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "@jongwooo",
        "name": "twitter:creator",
      },
      Object {
        "content": "All posts",
        "name": "twitter:title",
      },
      Object {
        "content": "TypeScript version of the default Gatsby starter",
        "name": "twitter:description",
      },
    ]
  }
  title="All posts"
  titleTemplate="%s | Library of Open Source Hardware"
/>
`;
